tests/integrationv2/test_client_authentication.py

```diff
@@ -231,3 +231,58 @@ def test_client_auth_with_s2n_client_with_cert(managed_process, provider, other_
         assert b'read client certificate' in results.stderr
         assert b'read certificate verify' in results.stderr
         assert_openssl_handshake_complete(results)
+
+
+"""
+In TLS 1.3, RSA-PSS is the recommended signature algorithm for RSA certificates, and the protocol mandates its use.
+However, not all servers support RSA-PSS signatures, particularly those built with openSSL1.0.2 versions. When the 
+server requests client authentication and the client sends an RSA certificate using TLS 1.3, if the server does not 
+support RSA-PSS, the connection fails. To avoid this, the client and server should negotiate a downgrade to TLS1.2.
+"""
+
+
+def test_tls_12_client_auth_downgrade(managed_process):
+    port = next(available_ports)
+
+    random_bytes = data_bytes(64)
+    client_options = ProviderOptions(
+        mode=Provider.ClientMode,
+        port=port,
+        data_to_send=random_bytes,
+        use_client_auth=True,
+        key=Certificates.RSA_2048_PKCS1.key,
+        cert=Certificates.RSA_2048_PKCS1.cert,
+        trust_store=Certificates.ECDSA_256.cert,
+        insecure=False,
+    )
+
+    client_options.extra_flags = ["--no-ca-verification"]
+
+    server_options = ProviderOptions(
+        mode=Provider.ServerMode,
+        port=port,
+        use_client_auth=True,
+        protocol=Protocols.TLS13,
+        key=Certificates.ECDSA_256.key,
+        cert=Certificates.ECDSA_256.cert,
+        trust_store=Certificates.RSA_2048_PKCS1.cert,
+        insecure=False,
+    )
+
+    server = managed_process(S2N, server_options, timeout=5)
+    client = managed_process(GnuTLS, client_options, timeout=5)
+
+    for results in client.get_results():
+        results.assert_success()
+
+    # If protocol version is TLS1.3 and lib-crypto is openssl-1.0.2, protocol version should downgrade to TLS1.2.
+    # The downgrade occurs because openssl-1.0.2 doesn't support RSA-PSS signature scheme.
+    if "openssl-1.0.2" in get_flag(S2N_PROVIDER_VERSION):
+        expected_protocol_version = Protocols.TLS12.value
+    else:
+        expected_protocol_version = Protocols.TLS13.value
+
+    for results in server.get_results():
+        results.assert_success()
+        assert to_bytes("Actual protocol version: {}".format(
+            expected_protocol_version)) in results.stdout
```

Comment:
I think it would be nice to add an assert here for what signature algorithm is used for client auth. Because then you could see that in TLS13 the sig alg is RSA PSS and in TLS12 the signature algorithm is just RSA.